data_structures/src/data_request.rs

```diff
@@ -359,7 +375,7 @@ mod tests {
         let dr_pointer = dr_transaction.hash();
 
         let mut p = DataRequestPool::default();
-        p.process_data_request(&dr_transaction, epoch);
+        let _a = p.process_data_request(&dr_transaction, epoch);
```

Comment:
The "unused result" warning means that the function could fail, so don't ignore that error

```suggestion
        p.process_data_request(&dr_transaction, epoch).unwrap();
```